### PR TITLE
[ISSUE #19]Message track query enhancement

### PR DESCRIPTION
**Pull request by zhangjidi2016**
## What is the purpose of the change

#19

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily. Notice, `it would be helpful if you could finish the following 5 checklist(the last one is not necessary)before request the community to review your PR`.

- [x] Make sure there is a [Github issue](https://github.com/apache/rocketmq/issues) filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue. 
- [x] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test(over 80% coverage) to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/apache/rocketmq/tree/master/test).
- [x] Run `mvn -B clean apache-rat:check findbugs:findbugs checkstyle:checkstyle` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.
- [ ] If this contribution is large, please file an [Apache Individual Contributor License Agreement](http://www.apache.org/licenses/#clas).

### CODE REVIEW

File: src/main/java/org/apache/rocketmq/dashboard/service/impl/TopicServiceImpl.java [hunk 1]
```diff
@@ -61,16 +61,28 @@
...
+    public TopicList fetchAllTopicList(boolean skipSysProcess, boolean skipRetryAndDlq) {
         try {
             TopicList allTopics = mqAdminExt.fetchAllTopicList();
             if (skipSysProcess) {
                 return allTopics;
             }
 
-            TopicList sysTopics = getSystemTopicList();
             Set<String> topics = new HashSet<>();
+            if (skipRetryAndDlq) {
```
Comment (StyleTang) [inline src/main/java/org/apache/rocketmq/dashboard/service/impl/TopicServiceImpl.java:72]: 1. What will happen if both skipSysProcess and skipRetryAndDlq are true.
2. maybe we can handle skipSysProcess and skipRetryAndDlq together, for example use Java lambda stream.
```java
allTopics.getTopicList().stream().map((item)->{
                ...
            }).filter(new Predicate<String>() {
                @Override
                public boolean apply(@Nullable String input) {
                    ...
                }
            }).collect(Collectors.toSet());
```

**Review by StyleTang — COMMENTED** [review]
Great, this feature helps users who use custom track message topic to query message trace. 👍  
I have several comments here.

1. message track topic should have a default value, after this PR, each user will need at least 3 clicks when making a query message trace query (even if they use the default message track topic).
2. I suggest that the message track topic should be in the message trace page but not in MsgTraceDetailDialog, and the browser can cache which topic was chosen before.
3. rocketmq.config.msgTrackTopicName is useless, we can just remove it.

File: src/main/java/org/apache/rocketmq/dashboard/controller/MessageTraceController.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ public Object viewTraceMessages(@RequestParam String msgId) {
 
     @RequestMapping(value = "/viewMessageTraceGraph.query", method = RequestMethod.GET)
     @ResponseBody
-    public MessageTraceGraph viewMessageTraceGraph(@RequestParam String msgId) {
-        return messageTraceService.queryMessageTraceGraph(msgId);
+    public MessageTraceGraph viewMessageTraceGraph(@RequestParam String msgId, @RequestParam String traceTopic) {
```
[the remark this answers is not in the document]
Comment (zhangjidi2016) [reply src/main/java/org/apache/rocketmq/dashboard/controller/MessageTraceController.java:67]: TraceTopic can be empty and system topic (RMQ_SYS_TRACE_TOPIC) is used by default.

File: src/main/java/org/apache/rocketmq/dashboard/service/impl/TopicServiceImpl.java (hunk 1 above)
Comment (zhangjidi2016) [reply src/main/java/org/apache/rocketmq/dashboard/service/impl/TopicServiceImpl.java:72]: Yes,  need to take into account the fact that both skipSysProcess and skipRetryAndDlq are true，I will optimize this method using stream.

**Comment by StyleTang** [comment]
In my point of view, I don't suggest putting the topic selection on the popover. It’s more appropriate to put it on the page.
If we put it on the message trace page, for most users who use the default message track topic, they can just ignore it, but on the popover, the unnecessary click is required.


image

I suggest something like below 
image

**Review by StyleTang — APPROVED** [review]

**Pull request merged**